psp/data/data_sources/nwp.py

```diff
@@ -109,10 +110,15 @@ def __init__(
         y_is_ascending: Is the `y` coordinate in ascending order. If it's in descending order, set
             this to `False`.
         """
-        if isinstance(paths, str):
-            paths = [paths]
-
-        self._paths = paths
+        if isinstance(paths_or_data, str):
+            paths_or_data = [paths_or_data]
+
+        if isinstance(paths_or_data, xr.Dataset):
```

Comment:
👍🏻

Reply:
Mostly useful for tests! We have the same in the PV data source.